### PR TITLE
AP_SmartRTL: give warning buffer is about to fill and de-activate

### PR DESCRIPTION
**Pull request by Hwurzburg**
tested in SITL....tried 90%, 20 points....10 points seemed a good compromise letting one stop and start SmartRTL before running out of buffer, but not nagging when 50 points get pruned often....20 points was good also

### CODE REVIEW

**Comment by rmackay9** [comment]
Might need to add a variable (boolean or time-since-last-send) so that it doesn't send the message too often to the GCS.

**Comment by Hwurzburg** [comment]
yep....but with 10 points you dont get a lot of time before the buffer fills, so not much nagging...have to hit rtl pretty quickly or overflow...still possible to stop and start back however...

File: libraries/AP_SmartRTL/AP_SmartRTL.cpp [hunk 1]
```diff
@@ -330,7 +330,7 @@ void AP_SmartRTL::run_background_cleanup()
     const uint16_t path_points_completed_limit = _path_points_completed_limit;
     _path_points_completed_limit = SMARTRTL_POINTS_MAX;
     _path_sem.give();
-
```
Comment (rmackay9) [inline libraries/AP_SmartRTL/AP_SmartRTL.cpp]: slight issue with the formatting as well ([style guide](http://ardupilot.org/dev/docs/style-guide.html)).  We always use curly brackets even if there's only a single statement afterwards.

Comment (Hwurzburg) [reply libraries/AP_SmartRTL/AP_SmartRTL.cpp]: was wondering about that....will fix

Comment (Hwurzburg) [reply libraries/AP_SmartRTL/AP_SmartRTL.cpp]: fixed

**Comment by Hwurzburg** [comment]
added a 30 sec do not repeat interval....

**Comment by Hwurzburg** [comment]
accidentally pushed to the wrong branch

**Comment by Hwurzburg** [comment]
after last boat test, decided to change it a little....moved it down AFTER simplification is completed, and lowered the remaining slot count below the minimum space that triggers simplification, since this gets triggered often on long straight runs and well before space is completely depleted.....now only warns if it cannot be compacted below minimum space level....with this reduction in warnings to near the very end of buffer space, a 10sec repeat only gives a couple of warnings before de-activation  occurs in STIL....seems about right now...

**Comment by Hwurzburg** [comment]
Boat test...  https://youtu.be/0yDlNFhYsWE
..a faster vehicle will give less warning time between first warning and de-activation...on a 2mph boat, this is about as much nagging as you will ever get...in SITL on a fast rover, it gives only about two warnings before de-activation occurs...

**Comment by rmackay9** [comment]
Merged after making a non-functional change which was to replace the now variable with a call to AP_HAL::millis() because it's only used once.

**Pull request closed without merging**